lib/tcpip/BufferManagement.cc

```diff
@@ -10,6 +10,23 @@
 
 using Debug = ConditionalDebug<false, "Buffer management">;
 
+// Use a separate allocator quota for the buffer manager (false by default).
+// The buffer manager is responsible for allocating network buffers, which
+// differs from the other types of allocations the TCP/IP stack performs. It
+// may thus make sense to give it its own quota. We may want to expose this as
+// a build system configuration option at some point.
+#define USE_DEDICATED_BUFFERMANAGER_POOL false
+// Size of the buffer manager allocator quota. Only relevant if
+// USE_DEDICATED_BUFFERMANAGER_POOL is enabled.
+#define BUFFER_MANAGER_QUOTA (ipconfigNUM_NETWORK_BUFFER_DESCRIPTORS * (ipconfigTCP_MSS + ipBUFFER_PADDING))
+
+#if USE_DEDICATED_BUFFERMANAGER_POOL
+DECLARE_AND_DEFINE_ALLOCATOR_CAPABILITY(BMMalloc, BUFFER_MANAGER_QUOTA);
+#define BM_MALLOC STATIC_SEALED_VALUE(BMMalloc)
```

Comment:
Please name this as quota or capability. I would assume that something called foo malloc is a malloc for foo, not an object passed to an allocator.